src/types/rule.ts

```diff
@@ -80,9 +80,21 @@ export interface IRulePatternOptions {
 }
 export type PatternRule = IRule<RuleFunction.PATTERN, IRulePatternOptions>;
 
+export enum CasingType {
```

Comment:
Do we want stuff spilling over into this rule file? This seems like a layover from when the core rulset and the core were completely indistinguishable, but now that we have custom functions, does anything other than the casing function need access to this? 

If not, maybe we can do a bit of tidy up and start moving function specific types, enums, interfaces, etc into the relevant function files to avoid further sprawl?

Reply:
Fixed